flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/RelShuttles.scala

```diff
@@ -97,14 +97,15 @@ class ExpandTableScanShuttle extends RelShuttleImpl {
   }
 
   /**
-    * Converts [[LogicalTableScan]] the result [[RelNode]] tree by calling [[RelTable]]#toRel
+    * Converts [[LogicalTableScan]] the result [[RelNode]] tree
+    * by calling [[QueryOperationCatalogViewTable]]#toRel
     */
   override def visit(scan: TableScan): RelNode = {
     scan match {
       case tableScan: LogicalTableScan =>
-        val relTable = tableScan.getTable.unwrap(classOf[RelTable])
-        if (relTable != null) {
-          val rel = relTable.toRel(RelOptUtil.getContext(tableScan.getCluster), tableScan.getTable)
+        val viewTable = tableScan.getTable.unwrap(classOf[QueryOperationCatalogViewTable])
```

Comment:
AbstractCatalogView from Catalog has no QueryOperation. We must get the underlie QueryOperation and convert to QueryOperationCatalogViewTable somewhere.

Reply:
thanks for your suggestion @docete , I have add a `TODO` in DatabaseCalciteSchema, and will support expanding `GenericCatalogView` later

Reply:
This is something that has to be done also for the legacy planner. The problem is that we have to pass somehow the sql parser, as the CatalogView translates to a string query.

Reply:
yes, we meet this problem in blink code. and we put the `FlinkPlannerImpl` instance into the calcite planner context, and get it from context to parser a string query in `CatalogTableToTableSourceRule`. it's a little hack! We will try to find a clean way to resolve it  in next pr.